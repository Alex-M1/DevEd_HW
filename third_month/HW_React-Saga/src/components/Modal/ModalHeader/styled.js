import styled from 'styled-components';

export const StModalHeader = styled.div`
  display: flex;
  justify-content: space-between;

  span{
    font-size: 26px;
    font-weight: bold;
    cursor: pointer;
  }
`;
