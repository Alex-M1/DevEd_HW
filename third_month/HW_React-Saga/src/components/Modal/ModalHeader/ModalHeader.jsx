import React from 'react';
import PropTypes from 'prop-types';
import { StModalHeader } from './styled';

function ModalHeader({ closeModal }) {
  return (
    <StModalHeader>
      <h1>Add person</h1>
      <span onClick={closeModal}>тип</span>
    </StModalHeader>
  );
}

export default ModalHeader;

ModalHeader.propTypes = {
  closeModal: PropTypes.func.isRequired,
};
