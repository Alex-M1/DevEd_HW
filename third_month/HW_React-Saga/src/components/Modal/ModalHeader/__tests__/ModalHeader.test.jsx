import React from 'react';
import ModalHeader from '../ModalHeader';

describe('ModalHeader', () => {
  const props = {
    closeModal: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = global.shallowSmart(<ModalHeader {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StModalHeader', () => {
    const component = global.mountSmart(<ModalHeader {...props} />);
    expect(component.find('styled__StModalHeader')).toHaveLength(1);
  });
  it('Should have h1', () => {
    const component = global.mountSmart(<ModalHeader {...props} />);
    expect(component.find('h1')).toHaveLength(1);
  });
  it('Should have span', () => {
    const component = global.mountSmart(<ModalHeader {...props} />);
    expect(component.find('span')).toHaveLength(1);
  });
  it('Should have click close modal', () => {
    const component = global.mountSmart(<ModalHeader {...props} />);
    component.find('span').simulate('click');
    expect(component.props().closeModal).toHaveBeenCalled();
  });
});
