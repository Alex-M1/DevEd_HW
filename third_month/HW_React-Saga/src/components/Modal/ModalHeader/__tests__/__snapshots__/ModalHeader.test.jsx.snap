// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader Should match snapshot 1`] = `"<div class=\\"styled__StModalHeader-sc-1ozqfyh-0 jmUcbk\\"><h1>Add person</h1><span>тип</span></div>"`;
